src/DssEmergencySpell.sol

```diff
@@ -0,0 +1,135 @@
+    function officeHours() external view override(DssExec, DssAction) returns (bool);
+}
+
+abstract contract DssEmergencySpell is DssEmergencySpellLike {
+    /// @dev The chainlog contract reference.
+    ChainlogLike internal constant _log = ChainlogLike(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
+
+    // @dev The reference to the `pause` contract.
+    address public immutable pause = _log.getAddress("MCD_PAUSE");
+    /// @dev The chainlog address.
+    address public constant log = address(_log);
+    /// @dev In regular spells, `eta` is used to enforce the GSM delay.
+    ///      For emergency spells, the GSM delay is not applicable.
+    uint256 public constant eta = 0;
+    /// @dev Keeping the same value as regular spells.
+    bytes public constant sig = abi.encodeWithSelector(DssAction.execute.selector);
+    /// @dev Emergency spells should not expire.
+    uint256 public constant expiration = type(uint256).max;
+    // @dev Office Hours is always `false` for emergency spells.
+    bool public constant officeHours = false;
+    // @dev `action` is expected to return a valid address.
+    //      We also implement the `DssAction` interface in this contract.
+    address public immutable action = address(this);
+    // @dev An emergency spell can be cast as soon as it is deployed.
+    //      Notice that cast is always a no-op.
+    uint256 internal immutable _nextCastTime = block.timestamp;
+
+    /**
+     * @dev In regular spells, `tag` is an immutable variable with the code hash of the spell action.
+     *      It specifically uses a separate contract for spell action because `tag` is immutable and the code hash of
+     *      the contract being initialized is not accessible in the constructor.
+     *      Since we do not have a separate contract for actions in Emergency Spells, `tag` has to be turned into a
+     *      getter function instead of an immutable variable.
+     * @return The contract codehash.
+     */
+    function tag() external view returns (bytes32) {
+        return address(this).codehash;
+    }
+
+    /**
+     * @notice Triggers the emergency actions of the spell.
+     * @dev Emergency spells are triggered when scheduled.
+     *      This function maintains the name for compatibility with regular spells, however nothing is actually being
+     *      scheduled. Emergency spells take affect immediately, so there is no need to call `pause.plot()`.
+     */
+    function schedule() external {
+        _emergencyActions();
+    }
+
+    /**
+     * @notice Implements the emergency actions to be triggered by the spell.
+     */
+    function _emergencyActions() internal virtual;
+
+    /**
+     * @notice Returns `_nextCastTime`.
+     * @dev this function exists only to keep interface compatibility with regular spells.
+     */
+    function nextCastTime() external view returns (uint256 castTime) {
+        return _nextCastTime;
+    }
```

Comment:
Nit: if we don't need nor expect `cast` to be triggered, would it make more sense to return `type(uint256).max`? It shouldn't make a difference in practice, since it will never be `plan`ned into pause contract (unless we change this logic based on the previous comment)